AndroidManifest.xml

```diff
@@ -12,6 +12,8 @@
     <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
 
+    <uses-permission android:name="com.android.launcher.permission.INSTALL_SHORTCUT" />
```

Comment:
As far as I see it, this should work without user interaction, but I will just test later by "manually" installing the apk.
I would offer it as an alternative to the widget, but if you think the user experience is better with just one option this would also be ok for me.